src/app/util/ember-compatibility-functions.cpp

```diff
@@ -201,6 +201,23 @@ CHIP_ERROR ReadSingleClusterData(ClusterInfo & aClusterInfo, TLV::TLVWriter * ap
                   ChipLogValueMEI(aClusterInfo.mClusterId), ChipLogValueX64(aClusterInfo.mNodeId), aClusterInfo.mEndpointId,
                   aClusterInfo.mFieldId, aClusterInfo.mListIndex);
 
+    AttributeAccessInterceptor * interceptor = findAttributeAccessInterceptor(aClusterInfo.mEndpointId, aClusterInfo.mClusterId);
+    if (interceptor != nullptr)
+    {
+        bool dataRead;
+        // TODO: We should probably clone the writer and convert failures here
+        // into status responses, unless our caller already does that.
+        ReturnErrorOnFailure(interceptor->ReadAttributeData(aClusterInfo, apWriter, &dataRead));
+
+        if (dataRead)
```

Comment:
If we use dataRead to signal the success or failure of the ReadAttributeData, what is its return value used for? Why not just use the return status to signal the result status?

Reply:
dataRead is not signaling success or failure.  It's signaling whether this was an attribute the interceptor thought it's supposed to handle or not.

We _could_ signal "I don't handle that attribute" that via a custom status code, but it's not really a "failure" condition to not be intercepting a particular attribute.

That said, I don't feel very strongly about this.